applications/nrf_desktop/src/modules/config.c

```diff
@@ -150,7 +159,41 @@ static void update_config(const u8_t config_id, u8_t *data, size_t size)
 	}
 }
 
-static int enable_settings(void)
+static int load_settings(void)
+{
+	int err = 0;
+
+	err = settings_load();
+
+	if (err) {
+		LOG_ERR("Cannot load settings");
+		module_set_state(MODULE_STATE_ERROR);
+	} else {
+		LOG_INF("Settings loaded");
+		module_set_state(MODULE_STATE_READY);
```

Comment:
I suggest moving setting module states to thread entry and setting state to ready after `err = load_settings();` at line 214.

This seems like a common code but it is only partially - setting state to error was removed in init (in case `settings_register` fails). It would actually be simpler to keep this two paths separated.

Reply:
changed